shell.nix

```diff
@@ -33,9 +33,9 @@ let
     in
     fenix.combine [
       nightly.rustfmt # should be the first
-      stable.rust
-      stable.rust-analyzer
-      stable.rust-src
+      nightly.rust
+      nightly.rust-analyzer
+      nightly.rust-src
```

Comment:
I believe we only want to use `rustfmt` from nightly.

cc @xzfc

Reply:
I'd be happy to merge if you remove these Nix changes, as they are unrelated to the benchmark problem.